packages/perseus/src/widgets/plotter.tsx

```diff
@@ -13,38 +13,17 @@ import KhanColors from "../util/colors";
 import GraphUtils from "../util/graph-utils";
 import KhanMath from "../util/math";
 
-import type {APIOptions, WidgetExports} from "../types";
+import type {PerseusPlotterWidgetOptions} from "../perseus-types";
+import type {WidgetExports, WidgetProps} from "../types";
 
 const {deepEq} = Util;
 
-const BAR = "bar";
-const LINE = "line";
-const PIC = "pic";
-const HISTOGRAM = "histogram";
-const DOTPLOT = "dotplot";
-
-const plotTypes = [BAR, LINE, PIC, HISTOGRAM, DOTPLOT];
-type PlotType = typeof plotTypes[number];
-
-type Props = {
-    type: PlotType;
-    labels: ReadonlyArray<string>;
-    categories: ReadonlyArray<string | number>;
-    scaleY: number;
-    maxY: number;
-    snapsPerLine: number;
-    picSize: number;
-    picBoxHeight: number;
-    picUrl: string;
-    plotDimensions: ReadonlyArray<number>;
-    labelInterval: number;
+type Rubric = PerseusPlotterWidgetOptions;
+type RenderProps = PerseusPlotterWidgetOptions;
 
-    starting: Array<number>;
-    static: boolean;
-    markings: any;
-    onChange: any;
-    trackInteraction: () => void;
-    apiOptions: APIOptions;
+type Props = WidgetProps<RenderProps, Rubric> & {
+    labelInterval: number;
+    picSize: number;
```

Comment:
I just checked.  These should be in the `PerseusPlotterWidgetOptions` also as they exist in our published content. 

Example: 

```json
{
    "options": {
      "scaleY": 1,
      "maxY": 10,
      "snapsPerLine": 2,
      "correct": [
        1
      ],
      "starting": [
        1
      ],
      "type": "bar",
      "labels": [
        "",
        ""
      ],
      "categories": [
        ""
      ],
      "picSize": 30,
      "picBoxHeight": 36,
      "picUrl": "https://cdn.kastatic.org/images/badges/earth-small.png",
      "plotDimensions": [
        275,
        200
      ],
      "labelInterval": 1
    },
    "type": "plotter",
    "version": {
      "major": 0,
      "minor": 0
    }
  }
```